numba/cuda/printimpl.py

```diff
@@ -69,18 +53,26 @@ def print_varargs(context, builder, sig, args):
     It dispatches to the appropriate 'print' implementations above
     depending on the detected real types in the signature."""
 
-    mod = builder.module
-    vprint = nvvmutils.declare_vprint(mod)
-    sep = context.insert_string_const_addrspace(builder, " ")
-    eol = context.insert_string_const_addrspace(builder, "\n")
+    vprint = nvvmutils.declare_vprint(builder.module)
+
+    formats = []
+    values = []
 
     for i, (argtype, argval) in enumerate(zip(sig.args, args)):
+        # XXX The return type is a lie: "print_item" doesn't return
```

Comment:
Will it be better to not use ``@lower("print_item", ...`` and use a custom singledispatch or other ways for dispatching?  Note that "print_item" is originally the implementation for the ``PRINT_ITEM`` bytecode.  I believe the interpreter doesn't generate calls to "print_item" anymore.